tmt/base.py

```diff
@@ -4326,8 +4326,8 @@ def resolve_dynamic_ref(
 
     # Read it, process it and get the value of the attribute 'ref'
     try:
-        with open(ref_filepath, encoding='utf-8') as datafile:
-            data = tmt.utils.yaml_to_dict(datafile.read())
+        data = tmt.utils.yaml_to_dict(ref_filepath.read_text(encoding='utf-8'))
+
```

Comment:
I believe utf8 is default, so could be dropped since we're "modernizing"?

Reply:
I’d love to, but I’m not sure about the default:

> … the contents of the file are returned as [str](https://docs.python.org/3/library/stdtypes.html#str), the bytes having been first decoded using a platform-dependent encoding or using the specified encoding if given.

It may not necessarily be `utf-8`, IIUIC. I’m not sure what’s better, whether we should leave it to OS and Python to pick a sane default, or whether we should enforce the default as we see it, or whether we should always enforce UTF-8, without any exception. I also think we still ignore the problem of locales when parsing output of various commands, too, so far we were able to get away with it. Seems like we have several topics to discuss :)

Reply:
Yeah, I believe another patch rather?